radix-engine/src/transaction/executor.rs

```diff
@@ -45,21 +61,34 @@ impl<'l, L: ReadableSubstateStore + WriteableSubstateStore> AbiProvider
     }
 }
 
-impl<'l, L: ReadableSubstateStore + WriteableSubstateStore> TransactionExecutor<'l, L> {
-    pub fn new(substate_store: &'l mut L, trace: bool) -> Self {
+impl<'s, 'w, S, W, I> TransactionExecutor<'s, 'w, S, W, I>
+where
+    S: ReadableSubstateStore + WriteableSubstateStore,
+    W: WasmEngine<I>,
+    I: WasmInstance,
+{
+    pub fn new(
+        substate_store: &'s mut S,
+        wasm_engine: &'w mut W,
+        trace: bool,
+    ) -> TransactionExecutor<'s, 'w, S, W, I> {
+        bootstrap(substate_store, wasm_engine);
```

Comment:
A bit weird to have bootstrap here

Reply:
Yep, I was trying to avoid bootstraping everywhere (which requires the wasm engine as in this PR). 

After more thoughts, I think it will make more sense if we:
- Remove `WasmEngine` in `Package` constructor, giving up validation cache since it's always billed. 
- Restore the `with_bootstrap` substate store constructor

Both changes will be done in https://github.com/radixdlt/radixdlt-scrypto/pull/316